internal/cmd/ocean/spark_connect.go

```diff
@@ -169,39 +174,45 @@ func (x *CmdSparkConnectOptions) Validate() error {
 	return nil
 }
 
-func (x *CmdSparkConnect) NewWebSocketServer() (*SocketServer, error) {
-
-	// todo Is there a some other way to get the env params from options?  Perhaps just use cmd options for all params
-	token := os.Getenv("SPOTINST_TOKEN")
-	account := os.Getenv("SPOTINST_ACCOUNT")
+func (x *CmdSparkConnect) connectToServer(ctx context.Context) (*SocketServer, error) {
+	cfg := spotinst.DefaultConfig()
+	cred, err := cfg.Credentials.Get()
+	if err != nil {
+		return nil, err
+	}
 
 	clusterID := x.opts.ClusterID
 	appID := x.opts.AppID
 	baseURL := x.opts.WsUrl
 
-	address := fmt.Sprintf("%s/ocean/spark/cluster/%s/app/%s/connect?accountId=%s", baseURL, clusterID, appID, account)
+	address := fmt.Sprintf("%s/ocean/spark/cluster/%s/app/%s/connect?accountId=%s", baseURL, clusterID, appID, cred.Account)
 	log.Infof("Starting websocket server on address %s", address)
 
-	header := http.Header{"Authorization": []string{"Bearer " + token}}
-	conn, resp, err := websocket.DefaultDialer.Dial(address, header)
+	header := http.Header{"Authorization": []string{"Bearer " + cred.Token}}
+	conn, resp, err := websocket.DefaultDialer.DialContext(ctx, address, header)
 
 	if err != nil {
-		if err == websocket.ErrBadHandshake {
+		if errors.Is(err, websocket.ErrBadHandshake) {
 			log.Errorf("handshake failed with status %d", resp.StatusCode)
 		}
 		return nil, err
 	}
-	x.startSocketServer(conn)
+
 	return &SocketServer{conn: conn}, nil
 }
 
-func (x *CmdSparkConnect) startSocketServer(wsConn *websocket.Conn) {
+func (x *CmdSparkConnect) startSocketServer(ctx context.Context, server SocketServer) {
```

Comment:
```suggestion
func (x *CmdSparkConnect) startSocketServer(ctx context.Context, server *SocketServer) {
```
I would suggest just passing the socket server as a pointer

Reply:
and yes, I noticed after last push the gobwas vendor was included as I was thinking about using it but stopped in middle of refactoring.  I will remove it